engines/asylum/views/menu.cpp

```diff
@@ -0,0 +1,2385 @@
+			updateQuitGame();
+			break;
+
+		case kMenuTextOptions:
+			updateTextOptions();
+			break;
+
+		case kMenuAudioOptions:
+			updateAudioOptions();
+			break;
+
+		case kMenuSettings:
+			updateSettings();
+			break;
+
+		case kMenuKeyboardConfig:
+			updateKeyboardConfig();
+			break;
+
+		case kMenuShowCredits:
+			updateShowCredits();
+			break;
+
+		case kMenuReturnToGame:
+			updateReturnToGame();
+			break;
+		}
+
+		// Ask for redraw
+		getSharedData()->setFlag(kFlagRedraw, true);
+	}
+
+	if (ticks > getSharedData()->getNextScreenUpdate()) {
+		if (getSharedData()->getFlag(kFlagRedraw)) {
+			getScreen()->copyBackBufferToScreen();
+
+			getSharedData()->setFlag(kFlagRedraw, false);
+			getSharedData()->setNextScreenUpdate(ticks + 55);
+		}
+	}
+
+	return true;
+}
+
+bool Menu::music() {
+	if (_activeScreen == kMenuShowCredits
+	 && _vm->isGameFlagSet(kGameFlagFinishGame)
+	 && !_dword_455D5C
+	 && !_dword_455D4C) {
+		_dword_455D5C = true;
+
+		getSound()->playMusic(kResourceNone, 0);
+		getSound()->playMusic(MAKE_RESOURCE(kResourcePackShared, 38));
+
+		return true;
+	}
+
+	return false;
+}
+
+bool Menu::key(const AsylumEvent &evt) {
+	switch (_activeScreen) {
+	default:
+		break;
+
+	case kMenuSaveGame:
+		keySaveGame(evt);
+		break;
+
+	case kMenuKeyboardConfig:
+		keyKeyboardConfig(evt);
+		break;
+
+	case kMenuShowCredits:
+		keyShowCredits();
+		break;
+	}
+
+	return true;
+}
+
+bool Menu::click(const AsylumEvent &evt) {
+	if (evt.type == Common::EVENT_RBUTTONDOWN
+	 && _activeScreen == kMenuShowCredits) {
+		 clickShowCredits();
+		 return true;
+	}
+
+	//////////////////////////////////////////////////////////////////////////
+	// Handle clicks on sub-screens
+	if (_activeScreen != kMenuNone) {
+		switch (_activeScreen) {
+		default:
+			break;
+
+		case kMenuNewGame:
+			clickNewGame();
+			break;
+
+		case kMenuLoadGame:
+			clickLoadGame();
+			break;
+
+		case kMenuSaveGame:
+			clickSaveGame();
+			break;
+
+		case kMenuDeleteGame:
+			clickDeleteGame();
+			break;
+
+		case kMenuViewMovies:
+			clickViewMovies();
+			break;
+
+		case kMenuQuitGame:
+			clickQuitGame();
+			break;
+
+		case kMenuTextOptions:
+			clickTextOptions();
+			break;
+
+		case kMenuAudioOptions:
+			clickAudioOptions();
+			break;
+
+		case kMenuSettings:
+			clickSettings();
+			break;
+
+		case kMenuKeyboardConfig:
+			clickKeyboardConfig();
+			break;
+
+		case kMenuShowCredits:
+			clickShowCredits();
+			break;
+
+		case kMenuReturnToGame:
+			clickReturnToGame();
+			break;
+		}
+
+		return true;
+	}
+
+	//////////////////////////////////////////////////////////////////////////
+	// Handle clicks on the main menu
+	_activeScreen = findMousePosition();
+	if (_activeScreen == kMenuNone)
+		return true;
+
+	getCursor()->set(MAKE_RESOURCE(kResourcePackShared, 3));
+	getText()->loadFont(kFontYellow);
+
+	switch (_activeScreen) {
+	default:
+		break;
+
+	case kMenuSaveGame:
+		_isEditingSavegameName = false;
+		// Fallback to next case
```

Comment:
Compiler Warning:
````
engines/asylum/views/menu.cpp: In member function ‘bool Asylum::Menu::click(const Asylum::AsylumEvent&)’:
engines/asylum/views/menu.cpp:647:26: warning: this statement may fall through [-Wimplicit-fallthrough=]
   _isEditingSavegameName = false;
   ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
engines/asylum/views/menu.cpp:650:2: note: here
  case kMenuLoadGame:
  ^~~~
````
Similar. You need // fallthrough to nudge the compiler to not warn here since the fallthrough does appear to be intentional.